zetasql/fuzzing/protobuf/zetasql_expression_extractor_test.cc

```diff
@@ -0,0 +1,267 @@
+//
+// Copyright 2020 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <memory>
+#include <string>
+
+#include "zetasql/fuzzing/protobuf/zetasql_expression_grammar.pb.h"
+#include "zetasql/fuzzing/protobuf/zetasql_expression_extractor.h"
+#include "gtest/gtest.h"
+
+using zetasql_expression_grammar::Expression;
+using zetasql_expression_grammar::LiteralExpr;
+using zetasql_expression_grammar::IntegerLiteral;
+using zetasql_expression_grammar::NumericLiteral;
+using zetasql_expression_grammar::CompoundExpr;
+using zetasql_expression_grammar::BinaryOperation;
+using parameter_grammar::Whitespace;
+using zetasql_fuzzer::ProtoExprExtractor;
+
+namespace zetasql_fuzzer {
+
+class ProtoExprExtractorTest : public ::testing::Test {};
+
+TEST_F(ProtoExprExtractorTest, ExtractorReleaseTest) {
```

Comment:
This is a weird test. I'd change it into three tests:

1. Builds an empty string if nothing is parsed.
EXPECT_EQ(extractor.GetResult(), "");

2. Parses an arbitrary string.
extractor.Parse("asd123");
EXPECT_EQ(extractor.GetResult(), "asd123");

3. Resets after GetResult.
extractor.Parse("dummy");
extractor.GetResult();
EXPECT_EQ(extractor.GetResult(), "dummy");

Reply:
For the third test we expect EXPECT_EQ(extractor.GetResult(), "");

Reply:
So per conversation yesterday:

Does it make more sense for the extractor to return an error if GetResult has already been called, assuming the model is (extract some number of times, get result, (clear) )?

Reply:
How should we return error, do we abort?